packages/@ember/object/compat.ts

```diff
@@ -148,25 +156,33 @@ export function dependentKeyCompat(
 
       assert(
         'The dependentKeyCompat() decorator must be passed a getter or setter when used in classic classes',
-        desc !== null &&
-          typeof desc === 'object' &&
-          (typeof desc.get === 'function' || typeof desc.set === 'function')
+        typeof desc.get === 'function' || typeof desc.set === 'function'
       );
 
       return wrapGetterSetter(target, key, desc!);
     };
 
     setClassicDecorator(decorator);
 
-    return decorator as Decorator;
+    return decorator;
   }
-
-  assert(
-    'The @dependentKeyCompat decorator must be applied to getters/setters when used in native classes',
-    (desc !== null && typeof desc!.get === 'function') || typeof desc!.set === 'function'
-  );
-
-  return wrapGetterSetter(target, key!, desc!);
 }
 
 setClassicDecorator(dependentKeyCompat as Decorator);
+
+function isPropertyDescriptor(value: unknown): value is PropertyDescriptor {
```

Comment:
should this have its own tests?, or are the existing tests sufficient?

Reply:
It's not public, so I don't think it needs its own tests.